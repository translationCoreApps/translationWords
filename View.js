/**
 * @description:
 *  This class defines the entire view for TranslationWords tool
 */
import React from 'react'
import {Row, Col, Tabs, Tab, Glyphicon} from 'react-bootstrap'
import CheckInfoCard from './components/CheckInfoCard.js'
import style from './css/style'
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider'

class View extends React.Component {
  render(){
    //Modules not defined within translationWords
    const { ScripturePane, VerseCheck, TranslationHelps } = this.props.modules;
    debugger;
    return (
      <MuiThemeProvider>
        <Row className="show-grid" style={{margin: '0px', bottom: "0px",  height: "100%"}}>
          <Col sm={12} md={6} lg={9} style={{height: "100%", padding: '0px'}}>
            <ScripturePane {...this.props} currentCheck={this.props.currentCheck} />
            <CheckInfoCard openHelps={this.props.toggleHelps} showHelps={this.props.showHelps} title={this.props.currentCheck.phrase} file={this.props.currentFile}/>
            <VerseCheck
              updateCheckStatus={this.props.updateCheckStatus}
              updateCurrentCheck={this.props.updateCurrentCheck}
              currentCheck={this.props.currentCheck}
              goToNext={this.props.goToNext}
              goToPrevious={this.props.goToPrevious}
              verse={this.props.targetVerse}
              direction={this.props.direction}
              bookName={this.props.bookName}
            />
          </Col>
          <Col sm={12} md={6} lg={3} style={{height: "100%", padding: "0px"}}>
            <div style={{height: "100vh"}}>
              <Glyphicon glyph={this.props.showHelps ? "chevron-right" : "chevron-left"}
                         style={this.props.showHelps ? style.tHelpsOpen : style.tHelpsClosed}
                         onClick={this.props.toggleHelps} />
                <div style={{display: this.props.showHelps ? "block" : "none", height: "100vh"}}>
                  <TranslationHelps currentFile={this.props.currentFile}
                                  online={this.props.online}/>
                </div>
            </div>
          </Col>
        </Row>
      </MuiThemeProvider>
    );
  }
}

module.exports = View;
