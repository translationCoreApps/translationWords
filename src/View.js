/**
 * @description:
 *  This class defines the entire view for TranslationWords tool
 */
import React from 'react';
import {Glyphicon} from 'react-bootstrap';
import CheckInfoCard from './components/CheckInfoCard.js';
import style from './css/style';
import PropTypes from 'prop-types';

class View extends React.Component {

  render() {
    // Modules not defined within translationWords
    const { ScripturePane, VerseCheck, TranslationHelps } = this.props.currentToolViews;

    // set the scripturePane to empty to handle react/redux when it first renders without required data
    let scripturePane = <div/>;
    // populate scripturePane so that when required data is preset that it renders as intended.
    if (this.props.settingsReducer.toolsSettings.ScripturePane !== undefined) {
      scripturePane = <ScripturePane {...this.props} />;
    }

    let { translationWords } = this.props.resourcesReducer.translationHelps;
    let { contextId } = this.props.contextIdReducer;
    let articleId = contextId.groupId;
    let currentFile;
    if (translationWords && translationWords[articleId]) {
      currentFile = this.props.resourcesReducer.translationHelps.translationWords[articleId];
    }

    return (
      <div style={{display: 'flex', flex: 'auto'}}>
        <div style={{flex: '2 1 900px', display: "flex", flexDirection: "column"}}>
          {scripturePane}
          <CheckInfoCard openHelps={this.props.toggleHelps} showHelps={this.props.showHelps} title={this.props.contextIdReducer.contextId.quote} file={currentFile}/>
          <VerseCheck {...this.props} />
        </div>
        <div style={{flex: this.props.showHelps ? '1 1 375px' : '0 0 30px', display: 'flex', justifyContent: 'flex-end', marginLeft: '-15px'}}>
          <div style={style.handleIconDiv}>
              <Glyphicon glyph={this.props.showHelps ? "chevron-right" : "chevron-left"}
                         style={style.handleIcon}
                         onClick={this.props.toggleHelps} />
          </div>
          <div style={{ display: this.props.showHelps ? "flex" : "none", flex: '1' }}>
            <TranslationHelps
              {...this.props}
              currentFile={currentFile}
              online={this.props.settingsReducer.online}
            />
          </div>
        </div>
      </div>
    );
  }
}

View.propTypes = {
  settingsReducer: PropTypes.object.isRequired,
  showHelps: PropTypes.func.isRequired,
  contextIdReducer: PropTypes.object.isRequired,
  toggleHelps: PropTypes.func.isRequired,
  resourcesReducer: PropTypes.object.isRequired,
  currentToolViews: PropTypes.object.isRequired
};

export default View;
