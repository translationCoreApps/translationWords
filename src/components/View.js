/* eslint-env jest */

/**
 * @description:
 *  This class defines the entire view for TranslationWords tool
 */

import React from 'react';
import {Glyphicon} from 'react-bootstrap';
import CheckInfoCard from './CheckInfoCard.js';
import style from '../css/style';
import PropTypes from 'prop-types';

class View extends React.Component {

  render() {
    // Modules not defined within translationWords
    const { ScripturePane, VerseCheck, TranslationHelps } = this.props.currentToolViews;
    const { title } = this.props;

    // set the scripturePane to empty to handle react/redux when it first renders without required data
    let scripturePane = <div></div>;
    // populate scripturePane so that when required data is preset that it renders as intended.
    if (this.props.settingsReducer.toolsSettings.ScripturePane !== undefined) {
      scripturePane = <ScripturePane {...this.props} />;
    }

    let { translationWords } = this.props.resourcesReducer.translationHelps;
    let { contextId } = this.props.contextIdReducer;
    let articleId = contextId.groupId;
    let currentFile;
    if (translationWords && translationWords[articleId]) {
      currentFile = this.props.resourcesReducer.translationHelps.translationWords[articleId];
    }

    return (
      <div style={{display: 'flex', flex: 'auto'}}>
        <div style={{flex: '2 1 900px', display: "flex", flexDirection: "column"}}>
          {scripturePane}
          <CheckInfoCard openHelps={this.props.toggleHelps} showHelps={this.props.showHelps} title={title} file={currentFile}/>
          <VerseCheck {...this.props} />
        </div>
        <div style={{flex: this.props.showHelps ? '1 1 375px' : '0 0 30px', display: 'flex', justifyContent: 'flex-end', marginLeft: '-15px'}}>
          <div style={style.handleIconDiv}>
              <Glyphicon glyph={this.props.showHelps ? "chevron-right" : "chevron-left"} style={style.handleIcon} onClick={this.props.toggleHelps} />
          </div>
          <div style={{ display: this.props.showHelps ? "flex" : "none", flex: '1' }}>
            <TranslationHelps
              {...this.props}
              currentFile={currentFile}
              online={this.props.settingsReducer.online}
            />
          </div>
        </div>
      </div>
    );
  }
}

View.propTypes = {
  currentToolViews: PropTypes.shape({
    ScripturePane: PropTypes.any,
    VerseCheck: PropTypes.any,
    TranslationHelps: PropTypes.any
  }),
  settingsReducer: PropTypes.shape({
    toolsSettings: PropTypes.shape({
      ScripturePane: PropTypes.any
    }),
    online: PropTypes.bool
  }),
  resourcesReducer: PropTypes.shape({
    translationHelps: PropTypes.shape({
      translationWords: PropTypes.object
    })
  }),
  contextIdReducer: PropTypes.shape({
    contextId: PropTypes.shape({
      groupId: PropTypes.any.isRequired,
      quote: PropTypes.string.isRequired
    })
  }),
  title: PropTypes.string.isRequired,
  toggleHelps: PropTypes.any.isRequired,
  showHelps: PropTypes.any.isRequired
};

module.exports = View;
