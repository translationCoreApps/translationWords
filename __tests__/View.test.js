/* eslint-env jest */
import React from 'react';
import View from '../src/View';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import renderer from 'react-test-renderer';

jest.mock('../src/components/CheckInfoCard.js', () => '[CheckInfoCard]');

describe('View component Tests', () => {
  it('Check View component', () => {
    let props = {
      currentToolViews: {
        ScripturePane: () => '[ScripturePane]',
        VerseCheck: () => '[VerseCheck]',
        TranslationHelps: () => '[TranslationHelps]'
      },
      settingsReducer: {
        toolsSettings: {
          ScripturePane: {}
        },
        online: true
      },
      resourcesReducer: {
        translationHelps: {
          translationWords: {'group1': 'file'}
        }
      },
      contextIdReducer: {
        contextId: {
          groupId: 'group1',
          quote: 'title'
        }
      },
      toggleHelps: jest.fn(),
      showHelps: jest.fn()
    };
    const component = renderer.create(
      <MuiThemeProvider>
        <View {...props} />
      </MuiThemeProvider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});