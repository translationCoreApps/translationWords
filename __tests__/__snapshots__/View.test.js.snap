// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`View component Tests Check View component 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "flex": "auto",
    }
  }
>
  <div
    style={
      Object {
        "display": "flex",
        "flex": "2 1 900px",
        "flexDirection": "column",
      }
    }
  >
    [ScripturePane]
    <[CheckInfoCard]
      file="file"
      openHelps={[Function]}
      showHelps={[Function]}
      title="title"
    />
    [VerseCheck]
  </div>
  <div
    style={
      Object {
        "display": "flex",
        "flex": "1 1 375px",
        "justifyContent": "flex-end",
        "marginLeft": "-15px",
      }
    }
  >
    <div
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flex": "0 0 15px",
        }
      }
    >
      <span
        className="glyphicon glyphicon-chevron-right"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "var(--text-color-dark)",
            "borderRadius": "5px 0px 0px 5px",
            "color": "var(--reverse-color)",
            "padding": "10px 0px",
            "zIndex": "999",
          }
        }
      />
    </div>
    <div
      style={
        Object {
          "display": "flex",
          "flex": "1",
        }
      }
    >
      [TranslationHelps]
    </div>
  </div>
</div>
`;
